numpy/fft/_pocketfft.c

```diff
@@ -20,6 +20,25 @@
 #include <stdlib.h>
 
 #include "npy_config.h"
+
+#ifdef NPY_HAVE_NEON
+#include "lowlevel_strided_loops.h"
+#include "arm_neon.h"
+
+/* align var to alignment */
+#define LOOP_BLOCK_ALIGN_VAR(var, type, alignment)\
+    npy_intp i, peel = npy_aligned_block_offset(var, sizeof(type),\
+                                                alignment, n);\
+    for(i = 0; i < peel; i++)
+
+#define LOOP_BLOCKED(type, vsize)\
+    for(; i < npy_blocked_end(peel, sizeof(type), vsize, n);\
+            i += (vsize / sizeof(type)))
+
+#define LOOP_BLOCKED_END\
+    for (; i < n; i++)
+#endif
```

Comment:
```suggestion

```
No need to handle the offset of the non-aligned part, for the following reasons
- You're dealing with 64bit load, it will not gonna make a difference even with old machines and it will not worst to add new npyv intrinsics to handle alignment memory access for interleaving/deinterleaving.
- NEON/ASIMD doesn't have intrinsics to handle the aligned load.
- Except for stream write(non-temporal) in the case of x86, modern machines and compilers can effectively handle non-aligned
  memory access even with the lowest memory alignments(1/2/4).